src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -70,6 +70,129 @@ public static NuGetFramework Parse(string folderName, IFrameworkNameProvider map
             return framework;
         }
 
+        /// <summary>
+        /// Creates a NuGetFramework from individual components
+        /// </summary>
+        public static NuGetFramework ParseComponents(string targetFrameworkIdentifier, string targetFrameworkVersion, string targetFrameworkProfile, string targetPlatformIdentifier, string targetPlatformVersion)
+        {
+            return ParseComponents(targetFrameworkIdentifier, targetFrameworkVersion, targetFrameworkProfile, targetPlatformIdentifier, targetPlatformVersion, DefaultFrameworkNameProvider.Instance);
+        }
+
+        /// <summary>
+        /// Creates a NuGetFramework from individual components, using the given mappings
+        /// </summary>
+        public static NuGetFramework ParseComponents(string targetFrameworkIdentifier, string targetFrameworkVersion, string targetFrameworkProfile, string targetPlatformIdentifier, string targetPlatformVersion, IFrameworkNameProvider mappings)
+        {
+            if (targetFrameworkIdentifier == null)
+            {
+                throw new ArgumentNullException("targetFrameworkIdentifier");
+            }
+
+            if (mappings == null)
+            {
+                throw new ArgumentNullException("mappings");
+            }
+
+            NuGetFramework result = null;
+
+            if (!TryParseSpecialFramework(targetFrameworkIdentifier, out result))
+            {
+                string framework = null;
+                if (!mappings.TryGetIdentifier(targetFrameworkIdentifier, out framework))
+                {
+                    framework = targetFrameworkIdentifier;
+                }
+
+                var version = new Version(0, 0);
+                var platformVersion = new Version(0, 0);
+                string profile = null;
+
+                if (!string.IsNullOrEmpty(targetFrameworkVersion))
+                {
+                    if (targetFrameworkVersion.IndexOf('.') < 0)
+                    {
+                        targetFrameworkVersion += ".0";
+                    }
+
+                    if (!Version.TryParse(targetFrameworkVersion, out version))
+                    {
+                        throw new ArgumentException(string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.InvalidFrameworkVersion,
+                            targetFrameworkVersion));
+                    }
+                }
+
+                if (version.Major >= 5
+                    && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.Net, framework))
+                {
+                    framework = FrameworkConstants.FrameworkIdentifiers.NetCoreApp;
+                }
```

Comment:
The point of this, for me, is that people can just pass in `net` as the `TargetFramework`, which might help a bit with sweeping our sleight-of-hand with `NETCoreApp` under the rug, for the times when people do component-based tfm definition. I like it better this way but I can change it if you feel strongly about it?

Reply:
The method added in this PR is not for parsing `TargetFramework`, but for parsing the individual components after the SDK has already parsed `TargetFramework` and extracted it into the multiple components in multiple MSBuild properties. This method re-assembles it into a NuGetFramework object.

Therefore the mapping that `net5.0` == `.NETCoreApp, Version=v5.0`, and not `.NETFramework` happens when `NuGetFramework.Parse(targetFramework)` happens, not when this 5 `string` `NuGetFramework.ParseComponents` is called.

It's somewhat academic because `.NETFramework, Version=v5.0` does not and will never exist. But calling this method with `targetFrameworkIdentifier` = `net` is just wrong, this method parses the 5 components, not the single "short folder name" string, and if someone calls the method with `targetFrameworkIdentifier` == `.NETFramework` and `targetFrameworkVersion` == `v5.0`, well, they very explicitly asked for the thing that doesn't exist. On one hand it increases the chance that API callers get the thing they want, even if they're doing it in the wrong way. On the other hand it makes it harder for them to detect when they're doing doing things "correctly".